components/modal/modal-content.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import clickOutside from 'react-click-outside';
+import { defer } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, compose, createRef } from '@wordpress/element';
+import { focus } from '@wordpress/utils';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import withFocusReturn from '../higher-order/with-focus-return';
+import withFocusContain from '../higher-order/with-focus-contain';
+
+const ESC_KEY = 27;
```

Comment:
There is a utility inside `@wordpress/utils` named `keycodes` for this, with`keycodes.ESCAPE` being the one you want.